server/src/main/java/org/opensearch/index/mapper/BooleanFieldMapper.java

```diff
@@ -257,9 +267,78 @@ public DocValueFormat docValueFormat(@Nullable String format, ZoneId timeZone) {
             return DocValueFormat.BOOLEAN;
         }
 
+        @Override
+        public Query termQuery(Object value, QueryShardContext context) {
+            failIfNotIndexedAndNoDocValues();
+            Query query = new TermQuery(new Term(name(), indexedValueForSearch(value)));
+            if (boost() != 1f) {
+                query = new BoostQuery(query, boost());
+            }
+            if (isSearchable() && hasDocValues()) {
+                return new IndexOrDocValuesQuery(
+                    query,
+                    SortedNumericDocValuesField.newSlowExactQuery(name(), Values.TRUE.bytesEquals(indexedValueForSearch(value)) ? 1 : 0)
+                );
+            }
+            if (hasDocValues()) {
+                return SortedNumericDocValuesField.newSlowExactQuery(name(), Values.TRUE.bytesEquals(indexedValueForSearch(value)) ? 1 : 0);
+            }
+            return query;
+        }
+
+        @Override
+        public Query termsQuery(List<?> values, QueryShardContext context) {
+            failIfNotIndexedAndNoDocValues();
+            if (isSearchable() && hasDocValues()) {
+                Query query = new TermInSetQuery(name(), values.stream().map(this::indexedValueForSearch).toArray(BytesRef[]::new));
+                Query dvQuery = new TermInSetQuery(
+                    MultiTermQuery.DOC_VALUES_REWRITE,
+                    name(),
+                    values.stream().map(this::indexedValueForSearch).toArray(BytesRef[]::new)
+                );
+                return new IndexOrDocValuesQuery(query, dvQuery);
+            }
+            if (hasDocValues()) {
+                return new TermInSetQuery(
+                    MultiTermQuery.DOC_VALUES_REWRITE,
+                    name(),
+                    values.stream().map(this::indexedValueForSearch).toArray(BytesRef[]::new)
+                );
+            }
+            return new TermInSetQuery(name(), values.stream().map(this::indexedValueForSearch).toArray(BytesRef[]::new));
+        }
```

Comment:
I think for Boolean fields, both `termsQuery` and `rangeQuery` should be evaluated based on the candidate values as one of the following:

1. `values` or `range` doesn't include `true` or `false` -> return `MatchNoDocsQuery`
2. `values` or `range` includes exactly one of `true` or `false` -> delegate to `termQuery`
3. `values` or `range` includes both `true` and `false` -> return a `FieldExistsQuery`

Reply:
I agree with this logic, I understand why a termsQuery might exist for Boolean fields but I was confused why there is ever a rangeQuery. Why would you need a rangeQuery when you can just do a termsQuery?